### PR TITLE
grpc-java

### PR DESCRIPTION
**Pull request by JustinB-SH**
Initial implementation of the `grpc-java` library, creating a running Kotlin client and server.

Still to come:
* This generates from a local copy of the sample proto file.  It should generate off of the shared instance elsewhere in the project path.
* Consolidate the dependencies.  They're copied and reused.

### CODE REVIEW

File: kotlin/grpc-java/model/src/main/proto/fortune_teller_api.proto [hunk 1]
```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
```
Comment (benjigoldberg) [inline kotlin/grpc-java/model/src/main/proto/fortune_teller_api.proto]: Does this need to be duplicated?

File: kotlin/grpc-java/server/src/main/kotlin/com/spothero/grpcjava/server/FortuneTellerService.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    val luckyNumbers = (1..5)
+      .map {
+        random.nextInt(1, 100)
+      }
+
+    val luckyAnimalIndex = random.nextInt(1, GetFortuneResponse.Animal.values().size - 2)
+    val luckyAnimal = GetFortuneResponse.Animal.forNumber(luckyAnimalIndex)
+
+    return GetFortuneResponse.newBuilder()
+      .setFortune("You're going to die")
```
Comment (benjigoldberg) [inline kotlin/grpc-java/server/src/main/kotlin/com/spothero/grpcjava/server/FortuneTellerService.kt]: cheery

File: kotlin/grpc-java/client/src/main/kotlin/com/spothero/grpcjava/client/FortuneTellerClient.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      throw e
+    }
+  }
+
+  private fun buildFortuneRequestWithCar(): GetFortuneRequest =
+    GetFortuneRequest.newBuilder()
+      .setName("Requesting with a Car")
+      .setOptionalCar(
+        GetFortuneRequest.Car.newBuilder()
+          .setMake("Geo")
```
Comment (benjigoldberg) [inline kotlin/grpc-java/client/src/main/kotlin/com/spothero/grpcjava/client/FortuneTellerClient.kt]: excellent choice in vehicle

File: kotlin/grpc-java/model/src/main/proto/fortune_teller_api.proto (hunk 1 above)
Comment (JustinB-SH) [reply kotlin/grpc-java/model/src/main/proto/fortune_teller_api.proto]: For the long term, no.  I should be able to change the protoc plugin to reference the actual file.  I copied it as the fastest way to get something working.
I'm going to play with configuring the path in a future release.

File: kotlin/grpc-java/server/src/main/kotlin/com/spothero/grpcjava/server/FortuneTellerService.kt (hunk 2 above)
Comment (JustinB-SH) [reply kotlin/grpc-java/server/src/main/kotlin/com/spothero/grpcjava/server/FortuneTellerService.kt]: It's the only fortune I could think of that will eventually come true.

File: kotlin/grpc-java/model/src/main/proto/fortune_teller_api.proto (hunk 1 above)
Comment (wilsoniya) [reply kotlin/grpc-java/model/src/main/proto/fortune_teller_api.proto]: It should be as easy as changing this path: https://github.com/spothero/idl-test/blob/23a704296e54d9965714ff5a120b22ea50a93cc3/idl/prototool.yaml#L20

**Pull request merged**